### PR TITLE
New user profile UI in User Settings

### PR DESCRIPTION
**Pull request by dbkr**
This replaces the profile setting UI in the general user settings tab.
 * Uses new Edit In Place component
 * Uses new Toast component
 * Adds a 'toast context' to manage these toasts (yes, we also have other things in the app we call toasts, but I figure better to have the name match the compound component).
 * Splits out the CSS of Room and User profile settings as they're not really related (and now even less so). Apologies for this not being a separate PR, if I'd realised sooner it would have been.

In order to make this PR slightly less huge, this is an intermediate step and doesn't add the context menu yet. That will come in a later PR.

Requires https://github.com/element-hq/compound-web/pull/166
Requires https://github.com/element-hq/compound-web/pull/173

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

File: playwright/snapshots/register/register.spec.ts/email-prompt-linux.png
Comment (t3chguy) [inline playwright/snapshots/register/register.spec.ts/email-prompt-linux.png:1]: This seems to have no relation to the PR title?

File: playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png
Comment (t3chguy) [inline playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png:1]: This middle alignment looks odd, is it intended?

File: playwright/snapshots/settings/security-user-settings-tab.spec.ts/Security-user-settings-tab-with-posthog-enable-b5d89-csLearnMoreDialog-should-be-rendered-properly-1-linux.png
Comment (t3chguy) [inline playwright/snapshots/settings/security-user-settings-tab.spec.ts/Security-user-settings-tab-with-posthog-enable-b5d89-csLearnMoreDialog-should-be-rendered-properly-1-linux.png:1]: This seems to have no relation to the PR title?

File: res/css/views/settings/_UserProfileSettings.pcss [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019, 2020 The Matrix.org Foundation C.I.C.
```
Comment (t3chguy) [inline res/css/views/settings/_UserProfileSettings.pcss:2]: Hello time travelling Dave

File: test/components/views/settings/UserProfileSettings-test.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
```
Comment (t3chguy) [inline test/components/views/settings/UserProfileSettings-test.tsx:2]: 🕐

File: playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png
Comment (dbkr) [reply playwright/snapshots/settings/general-user-settings-tab.spec.ts/general-smallscreen-linux.png:1]: Yeah, it's what the designs wanted.

File: res/css/views/settings/_UserProfileSettings.pcss (hunk 1 above)
Comment (dbkr) [reply res/css/views/settings/_UserProfileSettings.pcss:2]: Well, it's just a copy of a file with no (significant) changes so the copyright should surely remain? I guess I'll add 2024 though.

**Review by t3chguy — APPROVED** [review]

**Pull request merged**